### PR TITLE
[Support][#28010] Update Location Layout 

### PR DESCRIPTION
**Pull request by matTroiCuaHieu**
https://edu-redmine.sun-asterisk.vn/issues/28010

### CODE REVIEW

File: app/views/locations/_location.html.erb [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+    <div class="table-item-header col-sm-12">
+      <div style="white-space: nowrap;"><%= "#{location.warehouse_name} (#{location.id})" %></div>
+    </div>
+    <div class="table-item-content row">
+        <div class="col-sm-8 table-item-status">
+          <div><%= location.requests.last.plane_code %></div>
+        </div>
+        <div class="col-sm-4 table-item-action">
+          <%= link_to new_request_path do %>
+            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%=t("locations.index.request")%>></i>
```
Comment (chinhhq-0603) [inline app/views/locations/_location.html.erb]:
```suggestion
            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%= t("locations.index.request") %>></i>

```

File: app/views/locations/_location.html.erb [hunk 2]
```diff
@@ -1,18 +1,33 @@
...
+    <div class="table-item-header col-sm-12">
+      <div style="white-space: nowrap;"><%= "#{location.warehouse_name} (#{location.id})" %></div>
+    </div>
+    <div class="table-item-content row">
+        <div class="col-sm-8 table-item-status">
+          <div><%= t ".status.empty" %></div>
+        </div>
+        <div class="col-sm-4 table-item-action">
+          <%= link_to new_request_path do %>
+            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%=t("locations.index.request")%>></i>
```
Comment (chinhhq-0603) [inline app/views/locations/_location.html.erb]:
```suggestion
            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%= t("locations.index.request") %>></i>

```

File: app/views/locations/_radar.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    <div class="table-item-header col-sm-12">
+      <div style="white-space: nowrap;"><%= "#{location.warehouse_name} (#{location.id})" %></div>
+    </div>
+    <div class="table-item-content row">
+        <div class="col-sm-8 table-item-status">
+          <div><%= location.requests.last.plane_code %></div>
+        </div>
+        <div class="col-sm-4 table-item-action">
+          <%= link_to new_request_path do %>
+            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%=t("locations.index.request")%>></i>
```
Comment (chinhhq-0603) [inline app/views/locations/_radar.html.erb]: tuong tu

File: app/views/locations/_radar.html.erb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    <div class="table-item-header col-sm-12">
+      <div style="white-space: nowrap;"><%= "#{location.warehouse_name} (#{location.id})" %></div>
+    </div>
+    <div class="table-item-content row">
+        <div class="col-sm-8 table-item-status">
+          <div><%= t ".status.empty" %></div>
+        </div>
+        <div class="col-sm-4 table-item-action">
+          <%= link_to new_request_path do %>
+            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%=t("locations.index.request")%>></i>
```
Comment (chinhhq-0603) [inline app/views/locations/_radar.html.erb]: tuong tu

File: app/views/locations/_ready.html.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    <div class="table-item-header col-sm-12">
+      <div style="white-space: nowrap;"><%= "#{location.warehouse_name} (#{location.id})" %></div>
+    </div>
+    <div class="table-item-content row">
+        <div class="col-sm-8 table-item-status">
+          <div><%= location.requests.last.plane_code %></div>
+        </div>
+        <div class="col-sm-4 table-item-action">
+          <%= link_to new_request_path do %>
+            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%=t("locations.index.request")%>></i>
```
Comment (chinhhq-0603) [inline app/views/locations/_ready.html.erb]: tuoong tu

File: app/views/locations/_ready.html.erb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    <div class="table-item-header col-sm-12">
+      <div style="white-space: nowrap;"><%= "#{location.warehouse_name} (#{location.id})" %></div>
+    </div>
+    <div class="table-item-content row">
+        <div class="col-sm-8 table-item-status">
+          <div><%= t ".status.empty" %></div>
+        </div>
+        <div class="col-sm-4 table-item-action">
+          <%= link_to new_request_path do %>
+            <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%=t("locations.index.request")%>></i>
```
Comment (chinhhq-0603) [inline app/views/locations/_ready.html.erb]: tuong tu

File: app/views/locations/index.html.erb [hunk 7]
```diff
@@ -1,15 +1,67 @@
...
+          <div class="table-item-header col-sm-12">
+            <div style="white-space: nowrap;"><%= "#{location.warehouse_name} (#{location.id})" %></div>
+          </div>
+          <div class="table-item-content row">
+              <div class="col-sm-8 table-item-status">
+                <div><%= location.requests.last.plane_code %></div>
+              </div>
+              <div class="col-sm-4 table-item-action">
+                <%= link_to new_request_path do %>
+                  <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%=t(".request")%>></i>
```
Comment (chinhhq-0603) [inline app/views/locations/index.html.erb]: tuong tu

**Review by chinhhq-0603 — APPROVED** [review]

File: app/controllers/locations_controller.rb [hunk 8]
```diff
@@ -1,5 +1,15 @@
 class LocationsController < ApplicationController
   def index
-    @locations = Location.page(params[:page]).per Settings.pagination.user_page
+    @locations = Location.page(params[:page]).per Settings.pagination.locations_page
+    @radars = get_radar
+    @readys = get_ready
+  end
+
+  def get_ready
+    Location.where(warehouse_id: Settings.warehouse.ready)
```
Comment (hieplt-0914) [inline app/controllers/locations_controller.rb:9]: viet thanh scope

File: app/controllers/locations_controller.rb [hunk 9]
```diff
@@ -1,5 +1,15 @@
...
+  end
+
+  def get_radar
+    Location.where(warehouse_id: Settings.warehouse.gara)
```
Comment (hieplt-0914) [inline app/controllers/locations_controller.rb:13]: Viet thanh scope

File: app/javascript/packs/locations/location.js [hunk 10]
```diff
@@ -1,3 +1,10 @@
-$(function () {
-  $('[data-toggle="tooltip"]').tooltip()
+$("#select").change(function(){
+  $(".table-content").remove()
+  if($("#select option:selected")[0].value == "1"){
```
Comment (hieplt-0914) [inline app/javascript/packs/locations/location.js:3]: 1,2 ,3 dua len constant

File: app/views/locations/index.html.erb [hunk 11]
```diff
@@ -1,15 +1,67 @@
...
-      <th scope="col"><%= t ".plane" %></th>
-      <th scope="col"><%= t ".action" %></th>
-    </tr>
-  </thead>
-  <%= render @locations%>
-</table>
-<%= paginate @locations, theme: "twitter-bootstrap-4" %>
+<form remote:true>
+  <select id="select" class="custom-select" style="max-width: 100px; min-height: 30px">
+    <option id="select1" value="1" selected><%= t ".all" %></option>
```
Comment (hieplt-0914) [inline app/views/locations/index.html.erb:5]: 1,2,3 Settings

File: app/views/locations/index.html.erb [hunk 12]
```diff
@@ -1,15 +1,67 @@
...
+    <option id="select2" value="2" ><%= t ".repair_area" %></option>
+    <option id="select3" value="3" ><%= t ".ready_area" %></option>
+  </select>
+</form>
+<script>
+  $("#select").change(function(){
+    $(".table-content").remove()
+    if($("#select option:selected")[0].value == "1"){
```
Comment (hieplt-0914) [inline app/views/locations/index.html.erb:13]: 1,2,3 settings

**Pull request merged**